### PR TITLE
Update DeveloperGuide with Schedule List Feature description

### PR DESCRIPTION
**Pull request by seanflyyy**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -77,7 +77,30 @@ The following activity diagram summarizes what happens when a teacher executes a
...
+In order to update the `UI` from original view of a single panel to the right panel, there was a
+need to modify the existing `UI`. As a result, `MainWindow.fxml` had to be modified.
+
+Since there is a need for the schedule list to be a constant panel, which is separate from the `PersonListPanel`,
+a `ScheduleListPanel` under `UI` had to be created along with a `ScheduleListCard` class also under `UI`. The
+`ScheduleListCard` will contain more specific information than `PersonCard`. 
+
+Correspondingly, `PersonListCard.fxml` and `PersonListPanel.fxml` had to be created too.
+
+Since the `UI` depends on the `Model`, there was a need to update hte model to now include a `UniqueScheduleList`.
```
Comment (cadencjk) [inline docs/DeveloperGuide.md:93]: Typo: "hte" should be "the"

**Review by cadencjk — APPROVED** [review]
Just a small typo but LGTM!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -207,6 +207,31 @@ The following activity diagram summarizes what happens when a teacher executes a
     2. Only can set the class to a date at most 1 week away.
   
 
+### Schedule List Feature
```
Comment (Echomo-Xinyu) [inline docs/DeveloperGuide.md:210]: A link to this section should be added to the table of content under `## implementation` section

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -207,6 +207,31 @@ The following activity diagram summarizes what happens when a teacher executes a
...
+This feature allows the user to be able to view a schedule on the right hand side of the panel.
+
+#### Implementation Details
+In order to update the `UI` from original view of a single panel to the right panel, there was a
+need to modify the existing `UI`. As a result, `MainWindow.fxml` had to be modified.
+
+Since there is a need for the schedule list to be a constant panel, which is separate from the `PersonListPanel`,
+a `ScheduleListPanel` under `UI` had to be created along with a `ScheduleListCard` class also under `UI`. The
+`ScheduleListCard` will contain more specific information than `PersonCard`.
```
Comment (Echomo-Xinyu) [inline docs/DeveloperGuide.md:219]: it should be `PersonCard` contains more specific information than `ScheduleListCard` like `MoneyOwed`, `MoneyPaid`, `RatesPerClass` right

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -207,6 +207,31 @@ The following activity diagram summarizes what happens when a teacher executes a
     2. Only can set the class to a date at most 1 week away.
   
 
+### Schedule List Feature
+This feature allows the user to be able to view a schedule on the right hand side of the panel.
+
+#### Implementation Details
+In order to update the `UI` from original view of a single panel to the right panel, there was a
```
Comment (Echomo-Xinyu) [inline docs/DeveloperGuide.md:214]: I am personally not very sure about how we should write our implementation details for the front end and feel like less reasoning should be included here? So I am envisioning something like this for the current paragraph:

`MainWindow.fxml` is modified to update the `UI` from the original single-panel view to include an additional right panel.

What do you think? @cadencjk @gnahzeus @sjoann

**Review by Echomo-Xinyu — APPROVED** [review]
I think the PR is ready to be merged after addressing the several concerns listed above.

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (cadencjk) [reply docs/DeveloperGuide.md:214]: I agree with @Echomo-Xinyu, the sentence he gave is more concise but yet holds the same meaning.

**Pull request merged**